client/src/app/layout.tsx

```diff
@@ -1,24 +1,67 @@
-import type { Metadata } from "next";
+"use client"; // Mark the file as a client component
+
 import { Inter } from "next/font/google";
+import { createContext, useState, useContext, ReactNode, FC } from 'react';
 import "./globals.css";
 
 const inter = Inter({ subsets: ["latin"] });
 
-export const metadata: Metadata = {
-  title: "Create Next App",
-  description: "Generated by create next app",
+
+
+// Define the shape of the context value
+interface ThemeContextType {
+  theme: 'light' | 'dark';
+  toggleTheme: () => void;
+}
+
+// Create the context with a default value
+const ThemeContext = createContext<ThemeContextType | undefined>(undefined);
```

Comment:
Use `**zustand**` for state management not context api. Reason zustand is so lightweight and efficient and give better rendering with less bundelling size. Currently our project is using zustand got to the folder named stores you can see. 

Also create this theme as themeStore in stores folder instead of layout. Reason layout should be Server component.